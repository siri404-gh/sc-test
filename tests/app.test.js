import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import { BrowserRouter } from 'react-router-dom'

import App from '../src/app';

describe('App', () => {
    let container;
    const fakeResponse = {
      name: "Joni Baez",
      age: "32",
      address: "123, Charming Avenue"
    };
  
    global.fetch = jest.fn().mockImplementation(() =>
      Promise.resolve({
        json: () => Promise.resolve(fakeResponse),
      })
    );
    
    beforeEach(() => {
        container = document.createElement('section');
        document.body.appendChild(container);
    });

    afterEach(() => {
        unmountComponentAtNode(container);
        container.remove();
        container = null;
    });

    it('Should render', async () => {
        await act(async () => {
            render(<App />, container);
        });
        expect(container).toMatchSnapshot();
    });

    it('Should set cones', async () => {
        
          await act(async () => {
            render(<App />, container);
          });
        
          expect(container).toMatchSnapshot();
        
          global.fetch.mockClear();
          delete global.fetch;
    });
});