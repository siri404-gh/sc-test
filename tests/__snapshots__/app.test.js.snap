// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render 1`] = `<section />`;

exports[`App Should set cones 1`] = `<section />`;
